### ISSUE TITLE
Out-of-bounds exception in io::gzstream

### ISSUE THREAD

**Issue by meldavis** [meta-toolkit/meta#156]
The underflow method in [meta/io/gzstream.cpp](https://github.com/meta-toolkit/meta/blob/master/src/io/gzstream.cpp#L47) addresses memory outside of the array bounds.  It generates an exception in Windows (MSVC).

The buffer size is 512 bytes.
This line reads in 512 bytes:
```C++
auto bytes = gzread(file_, &buffer_[0], static_cast<unsigned>(buffer_.size()));
```
Then this line addresses one byte past the buffer: 
```C++
setg(&buffer_[0], &buffer_[0], &buffer_[static_cast<std::size_t>(bytes)]);
```

**Comment by skystrife** [comment]
Oh goodness, what a silly bug. Thanks for catching this. The second line should read something like

```cpp
setg(&buffer_[0], &buffer_[0], &buffer_[0] + static_cast<std::size_t>(bytes));
```

Would you like to make a pull request with the fix?

**Comment by meldavis** [comment]
I'm sorry, my schedule, right now, prevents me from being a contributor.   I just wanted to save you the time researching such a minor bug.   Hopefully, when things settle down here and I can delve more deeply into using Meta, I can contribute.  Thanks!

**Comment by skystrife** [comment]
Fixed in v2.4.2.
